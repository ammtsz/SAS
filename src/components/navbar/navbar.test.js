import React from "react";
import { render, screen } from "@testing-library/react"
import Navbar from "./navbar.component";

describe("<Navbar />", () => {
    it("should show User name on Navbar", () => {
        const userName = "Peter"
        render(<Navbar user={userName}/>)
        expect(screen.getByText("Hello, Peter")).toBeInTheDocument()
        expect(screen.getByText("Exit")).toBeInTheDocument()
    })
    it("should show 'Fazer Login' on Navbar", () => {
        const userName = null
        render(<Navbar user={userName}/>)
        expect(screen.getByText("Login")).toBeInTheDocument()
    })
})