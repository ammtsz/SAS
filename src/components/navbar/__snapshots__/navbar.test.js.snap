// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Navbar /> Theme toggle button should render Navbar 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdfBwQ jIIKgP"
  >
    <h3
      class="sc-gsTCUz iyGGEo"
    >
      Teste Dev Frontend
    </h3>
    <div
      class="sc-dlfnbm cyDqmn"
    >
      <p
        class="sc-hKgILt llgplH"
      />
      <button
        class="sc-eCssSg iTEdMl"
      >
        Log in
      </button>
      <span
        class="sc-jSgupP gOGJce"
      >
        <div
          class="sc-gKsewC cMTlMO"
        >
          <label
            class="sc-iBPRYJ gAapIF"
          >
            <input
              checked=""
              class="sc-pFZIQ cjTSOr"
              id="button-switch--navbar-toggle-theme"
              type="checkbox"
            />
            <span
              class="sc-fubCfw kkjObP slider round"
            />
          </label>
          <label
            class="button-switch--label"
            for="button-switch--navbar-toggle-theme"
          >
            dark
          </label>
        </div>
      </span>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`Component <Navbar /> User is logged in should render Navbar 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdfBwQ jIIKgP"
  >
    <h3
      class="sc-gsTCUz iyGGEo"
    >
      Teste Dev Frontend
    </h3>
    <div
      class="sc-dlfnbm cyDqmn"
    >
      <p
        class="sc-hKgILt llgplH"
      >
        Hello, Peter
      </p>
      <button
        class="sc-eCssSg iTEdMl"
      >
        Log out
      </button>
      <span
        class="sc-jSgupP gOGJce"
      >
        <div
          class="sc-gKsewC cMTlMO"
        >
          <label
            class="sc-iBPRYJ gAapIF"
          >
            <input
              class="sc-pFZIQ cjTSOr"
              id="button-switch--navbar-toggle-theme"
              type="checkbox"
            />
            <span
              class="sc-fubCfw kkjObP slider round"
            />
          </label>
          <label
            class="button-switch--label"
            for="button-switch--navbar-toggle-theme"
          />
        </div>
      </span>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`Component <Navbar /> User is logged out and it's on login page should render Navbar 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdfBwQ jIIKgP"
  >
    <h3
      class="sc-gsTCUz iyGGEo"
    >
      Teste Dev Frontend
    </h3>
    <div
      class="sc-dlfnbm cyDqmn"
    >
      <p
        class="sc-hKgILt llgplH"
      />
      <button
        class="sc-eCssSg iTEdMl"
      >
        skip log in
      </button>
      <span
        class="sc-jSgupP gOGJce"
      >
        <div
          class="sc-gKsewC cMTlMO"
        >
          <label
            class="sc-iBPRYJ gAapIF"
          >
            <input
              class="sc-pFZIQ cjTSOr"
              id="button-switch--navbar-toggle-theme"
              type="checkbox"
            />
            <span
              class="sc-fubCfw kkjObP slider round"
            />
          </label>
          <label
            class="button-switch--label"
            for="button-switch--navbar-toggle-theme"
          />
        </div>
      </span>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`Component <Navbar /> User is logged out and on homepage should render Navbar 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdfBwQ jIIKgP"
  >
    <h3
      class="sc-gsTCUz iyGGEo"
    >
      Teste Dev Frontend
    </h3>
    <div
      class="sc-dlfnbm cyDqmn"
    >
      <p
        class="sc-hKgILt llgplH"
      />
      <button
        class="sc-eCssSg iTEdMl"
      >
        Log in
      </button>
      <span
        class="sc-jSgupP gOGJce"
      >
        <div
          class="sc-gKsewC cMTlMO"
        >
          <label
            class="sc-iBPRYJ gAapIF"
          >
            <input
              class="sc-pFZIQ cjTSOr"
              id="button-switch--navbar-toggle-theme"
              type="checkbox"
            />
            <span
              class="sc-fubCfw kkjObP slider round"
            />
          </label>
          <label
            class="button-switch--label"
            for="button-switch--navbar-toggle-theme"
          />
        </div>
      </span>
    </div>
  </nav>
</DocumentFragment>
`;
